prov/sm2/src/sm2_progress.c

```diff
@@ -214,6 +229,7 @@ void sm2_progress_recv(struct sm2_ep *ep)
 			break;
 
 		if (xfer_entry->proto == sm2_proto_return) {
+			handle_delivery_complete(ep, xfer_entry);
```

Comment:
Since we only call this once, I think I prefer to have this inline here for readability